File: Blog-Backend/src/models/user_model.js
const { Schema, model } = require("mongoose");
const bcrypt = require("bcrypt");
const uuid = require("uuid");

const userSchema = new Schema({
    id: { type: String, unique: true },
    fullName: { type: String, required: true, },
    email: { type: String, unique: true, required: true },
    password: { type: String, required: true },
    avatar: { type: String, default: "" },
    level: { type: String, default: "Beginner" },
    createdOn: { type: Date },
    updatedOn: { type: Date },
});

userSchema.pre('save', function (next) {
    this.id = uuid.v1();
    this.createdOn = new Date();
    this.updatedOn = new Date();
    const salt = bcrypt.genSaltSync(10);
    const hash = bcrypt.hashSync(this.password, salt);
    this.password = hash;
    next();
});
userSchema.pre(['update', 'findOneAndUpdate', 'updateOne'], function (next) {
    const update = this.getUpdate();
    delete update._id;
    delete update.id;
    this.update = new Date();
    next();
});

const UserModel = model("User", userSchema);
module.exports = UserModel;

File: blog-frontend/blog-frontend/lib/data/models/user/user_model.dart
class UserModel {
  String? sId;
  String? fullName;
  String? email;
  String? password;
  String? avatar;
  String? level;
  String? id;
  String? createdOn;
  String? updatedOn;

  UserModel({
    this.sId,
    this.fullName,
    this.email,
    this.password,
    this.avatar,
    this.level,
    this.id,
    this.createdOn,
    this.updatedOn,
  });

  UserModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    fullName = json['fullName'];
    email = json['email'];
    password = json['password'];
    avatar = json['avatar'];
    level = json['level'];
    id = json['id'];
    createdOn = json['createdOn'];
    updatedOn = json['updatedOn'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.sId;
    data['fullName'] = this.fullName;
    data['email'] = this.email;
    data['password'] = this.password;
    data['avatar'] = this.avatar;
    data['level'] = this.level;
    data['id'] = this.id;
    data['createdOn'] = this.createdOn;
    data['updatedOn'] = this.updatedOn;
    return data;
  }
}
